src/importexport/musicxml/internal/musicxml/importmxmlpass1.cpp

```diff
@@ -1611,6 +1650,55 @@ void MusicXMLParserPass1::defaults()
     updateStyles(_score, wordFontFamily, wordFontSize, lyricFontFamily, lyricFontSize);
 }
 
+//---------------------------------------------------------
+//   setStyle
+//---------------------------------------------------------
+
+void MusicXMLParserPass1::setStyle(const QString& type, const double val)
+{
+    if (type == "light barline") {
+        _score->style().set(Sid::barWidth, Spatium(val / 10));
+    } else if (type == "heavy barline") {
+        _score->style().set(Sid::endBarWidth, Spatium(val / 10));
+    } else if (type == "beam") {
+        _score->style().set(Sid::beamWidth, Spatium(val / 10));
+    } else if (type == "bracket") {
+        _score->style().set(Sid::bracketWidth, Spatium(val / 10));
+    } else if (type == "dashes") {
+        _score->style().set(Sid::lyricsDashLineThickness, Spatium(val / 10));
+    } else if (type == "enclosure") {
+        _score->style().set(Sid::staffTextFrameWidth, Spatium(val / 10));
+    } else if (type == "ending") {
+        _score->style().set(Sid::voltaLineWidth, Spatium(val / 10));
+    } else if (type == "extend") {
+        _score->style().set(Sid::lyricsLineThickness, Spatium(val / 10));
+    } else if (type == "leger") {
+        _score->style().set(Sid::ledgerLineWidth, Spatium(val / 10));
+    } else if (type == "pedal") {
+        _score->style().set(Sid::pedalLineWidth, Spatium(val / 10));
+    } else if (type == "octave shift") {
+        _score->style().set(Sid::ottavaLineWidth, Spatium(val / 10));
+    } else if (type == "staff") {
+        _score->style().set(Sid::staffLineWidth, Spatium(val / 10));
+    } else if (type == "stem") {
+        _score->style().set(Sid::stemWidth, Spatium(val / 10));
+    } else if (type == "tuplet bracket") {
+        _score->style().set(Sid::tupletBracketWidth, Spatium(val / 10));
+    } else if (type == "wedge") {
+        _score->style().set(Sid::hairpinLineWidth, Spatium(val / 10));
+    } else if ((type == "slur middle") || (type == "tie middle")) {
+        _score->style().set(Sid::SlurMidWidth, Spatium(val / 10));
+    } else if ((type == "slur tip") || (type == "tie tip")) {
+        _score->style().set(Sid::SlurEndWidth, Spatium(val / 10));
+    } else if ((type == "cue")) {
+        _score->style().set(Sid::smallNoteMag, val / 100);
+    } else if ((type == "grace")) {
+        _score->style().set(Sid::graceNoteMag, val / 100);
+    } else if ((type == "grace-cue")) {
+        // not supported
```

Comment:
If cue and grace is set, we can probably reconstruct grace-cue from it, can't we?

Reply:
Yes, MuseScore does it automatically. But it's not possible the other way round, i.e. you don't know what values to set for `cue` and `grace` from `grace-cue`. We just have to ignore this value.